homeassistant/components/icloud/strings.json

```diff
@@ -6,7 +6,7 @@
         "description": "Enter your credentials",
         "data": {
           "username": "[%key:common::config_flow::data::email%]",
-          "password": "[%key:common::config_flow::data::password%]",
+          "password": "App-specific password",
```

Comment:
I just added a PR with app specific passwords as a translation string:
https://github.com/home-assistant/core/pull/121021

If that merges, or you do it yourself in this PR, you can use `[%key:common::config_flow::data::app_specific_password%]`

Reply:
I don't understand the relationship and why that is needed? Can you clarify?

Reply:
This is a translatable string... hence it is in `strings.json` and not in code...

Reply:
I'm having one of those days, doh!